File: projects/askfin_android/Models/QuestionRepository.kt
interface QuestionRepository {
    suspend fun getQuestions(): Result<List<Question>>
    suspend fun getQuestionsByCategory(
        category: QuestionCategory
    ): Result<List<Question>>
    suspend fun getWeakQuestions(
        userId: String,
        threshold: Int = 3
    ): Result<List<Question>>
}

// Usage (safe):
questionRepository.getQuestions()
    .onSuccess { questions -> updateUI(questions) }
    .onFailure { error -> showErrorMessage("Failed to load questions") }

File: projects/askfin_web/src/types/QuestionRepository.ts
import { Question } from './Question';
import { QuestionCategory } from './Answer';
export interface QuestionRepository {
  getQuestions(limit?: number): Promise<Question[]>;
  getQuestionsByCategory(
    category: QuestionCategory,
    limit?: number
  ): Promise<Question[]>;
  getWeakQuestions(
    userAnswers: UserAnswer[],
    options?: { limit?: number; threshold?: number }
  ): Promise<Question[]>;
}